backend/plugin/parser/mysql/affected_rows.go

```diff
@@ -0,0 +1,443 @@
+								case *mysql.CreateEventContext:
+									result = "CREATE_EVENT"
+								case *mysql.CreateTriggerContext:
+									result = "CREATE_TRIGGER"
+								case *mysql.CreateFunctionContext:
+									result = "CREATE_FUNCTION"
+								case *mysql.CreateProcedureContext:
+									result = "CREATE_PROCEDURE"
+								}
+							}
+						case *mysql.DropStatementContext:
+							for _, child := range ctx.GetChildren() {
+								switch child.(type) {
+								case *mysql.DropIndexContext:
+									result = "DROP_INDEX"
+								case *mysql.DropTableContext:
+									result = "DROP_TABLE"
+								case *mysql.DropDatabaseContext:
+									result = "DROP_DATABASE"
+								case *mysql.DropViewContext:
+									result = "DROP_VIEW"
+								case *mysql.DropTriggerContext:
+									result = "DROP_TRIGGER"
+								case *mysql.DropEventContext:
+									result = "DROP_EVENT"
+								case *mysql.DropFunctionContext:
+									result = "DROP_FUNCTION"
+								case *mysql.DropProcedureContext:
+									result = "DROP_PROCEDURE"
+								}
+							}
+						case *mysql.AlterStatementContext:
+							for _, child := range ctx.GetChildren() {
+								switch child.(type) {
+								case *mysql.AlterTableContext:
+									result = "ALTER_TABLE"
+								case *mysql.AlterDatabaseContext:
+									result = "ALTER_DATABASE"
+								case *mysql.AlterViewContext:
+									result = "ALTER_VIEW"
+								case *mysql.AlterEventContext:
+									result = "ALTER_EVENT"
+								}
+							}
+						case *mysql.TruncateTableStatementContext:
+							result = "TRUNCATE"
+						case *mysql.RenameTableStatementContext:
+							result = "RENAME"
+
+						// dml.
+						case *mysql.DeleteStatementContext:
+							result = "DELETE"
+						case *mysql.InsertStatementContext:
+							result = "INSERT"
+						case *mysql.UpdateStatementContext:
+							result = "UPDATE"
+						}
+					}
+				default:
+				}
+			}
+		default:
+		}
+	}
+	return result
+}
+
+type AffectedRowsListener struct {
+	*mysql.BaseMySQLParserListener
+
+	ctx          context.Context
+	sqlDB        *sql.DB
+	text         string
+	metadata     *storepb.DatabaseSchemaMetadata
+	affectedRows int64
+	err          error
+}
+
+// GetAffectedRows return the rows count affected by the sql.
+func GetAffectedRows(ctx context.Context, sqlDB *sql.DB, metadata *storepb.DatabaseSchemaMetadata, stmt *ParseResult) (int64, error) {
+	listener := &AffectedRowsListener{
+		ctx:      ctx,
+		sqlDB:    sqlDB,
+		metadata: metadata,
+	}
+	antlr.ParseTreeWalkerDefault.Walk(listener, stmt.Tree)
+	if listener.err != nil {
+		return 0, listener.err
+	}
+
+	return listener.affectedRows, nil
+}
+
+func (l *AffectedRowsListener) EnterQuery(ctx *mysql.QueryContext) {
+	l.text = ctx.GetParser().GetTokenStream().GetTextFromRuleContext(ctx)
+}
+
+// EnterInsertStatement is called when production insertStatement is entered.
+func (l *AffectedRowsListener) EnterInsertStatement(ctx *mysql.InsertStatementContext) {
```

Comment:
Looks more complex than what it was in `statement_report_affected_rows.go`.

Reply:
I only want to check the outermost sql.